flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java

```diff
@@ -201,6 +203,12 @@ public TableSchema commitChanges(List<SchemaChange> changes) throws Exception {
                                 DataType newType =
                                         TableSchema.toDataType(
                                                 update.newLogicalType(), new AtomicInteger(0));
+                                checkState(
```

Comment:
The behavior which check the implicit cast could move before line 202.

Reply:
Done